### PR TITLE
Test `authSource`

### LINKED ISSUE #4392: Check database name during authentication
### What should be done?

Should we check user's database and command's `$db` or that's a part of authorization? If yes, we should do it. If not, we should remove references.

### Where?

See references to this issue.

### Definition of Done

TBD

- handler updated;
- all backends updated;
- unit tests added/updated;
- integration/compatibility tests added/updated;
- spot refactorings done;
- user documentation updated or an issue to create documentation created;
- something else?

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #4392.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/users/connection_test.go [hunk 1]
```diff
@@ -307,6 +308,181 @@ func TestAuthenticationOnAuthenticatedConnection(t *testing.T) {
...
+		}).Err()
+		require.NoError(t, err)
+
+		t.Cleanup(func() {
+			err = db.Client().Database(dbName).RunCommand(ctx, bson.D{{"dropAllUsersFromDatabase", int32(1)}}).Err()
+			require.NoError(t, err)
+		})
+	}
+
+	for name, tc := range map[string]struct {
```
Comment (rumyantseva) [inline integration/users/connection_test.go:438]: There is one case that I don't see in this or the previous test.
What if `authSource` is set as something else than a `db1`? 
E.g. the user is correct, the password is correct, but `authSource` is set as a non-existent database, or `authSource` database exists, but no users were created with this database.

**Review by rumyantseva — COMMENTED** [review]
Looks good overall, asked one question.

File: integration/users/connection_test.go [hunk 2]
```diff
@@ -307,6 +308,181 @@ func TestAuthenticationOnAuthenticatedConnection(t *testing.T) {
...
+	t.Parallel()
+
+	s := setup.SetupWithOpts(t, nil)
+	ctx, db := s.Ctx, s.Collection.Database()
+
+	for name, tc := range map[string]struct {
+		username         string
+		password         string
+		authenticationDB string
+		path             string
```
Comment (AlekSi) [inline integration/users/connection_test.go:321]: The path component of MongoDB URI is officially called "defaultauthdb". Having those two separate fields makes it more complicated.

I wonder if having only two fields (a full URI with ":%d" for port and "current database") will make things easier.

**Review by AlekSi — COMMENTED** [review]
In our dev setup, MongoDB authentication is disabled. Did we run those tests manually against MongoDB with auth enabled?

File: integration/users/connection_test.go (hunk 2 above)
Comment (chilagrow) [reply integration/users/connection_test.go]: The test case is easier to read, body is slightly more complicated but I think it's better

File: integration/users/connection_test.go [hunk 3]
```diff
@@ -307,6 +308,175 @@ func TestAuthenticationOnAuthenticatedConnection(t *testing.T) {
...
+			q := u.Query()
+			for k, v := range testURI.Query() {
+				q.Set(k, v[0])
+			}
+
+			u.RawQuery = q.Encode()
+
+			opts := options.Client().ApplyURI(u.String())
+
+			t.Log("connecting", u.String())
```
Comment (chilagrow) [inline integration/users/connection_test.go:381]:
```suggestion
			t.Log("Connecting", u.String())
```

**Review by rumyantseva — APPROVED** [review]
Looks good!

**Comment by AlekSi** [comment]
@chilagrow You are right, I confused `mongodb.conf` and `mongodb_secured.conf`

**Review by noisersup — APPROVED** [review]

**Comment by chilagrow** [comment]
Yeah, the dev environment setup is a bit too confusing and complicated...

**Review by AlekSi — APPROVED** [review]

**Pull request merged**